main.cc

```diff
@@ -1212,171 +1218,194 @@ int main(int argc, char*argv[])
       }
 
-	/* On with the process of elaborating the module. */
-      Design*des = elaborate(roots);
+      /* On with the process of elaborating the module. */
+
+      std::unique_ptr<Design> des(elaborate(roots));
 
-      if ((des == 0) || (des->errors > 0)) {
-	    if (des != 0) {
-		  cerr << des->errors
-		       << " error(s) during elaboration." << endl;
-		  if (net_path) {
-			ofstream out (net_path);
-			des->dump(out);
-		  }
-	    } else {
-		  cerr << "Elaboration failed" << endl;
-	    }
 
-	    goto errors_summary;
+      if(des == nullptr){
+            // elaborate failed in unexpected or unspecified way
+            cerr << "Elaboration failed" << endl;
+            EOC_cleanup();
+            return 1;
+      } else if(des->errors > 0) {
+            cerr << des->errors
+                  << " error(s) during elaboration." << endl;
+            // Elaboration resulted in errors
+            // This function call ultimately kills the program, therefore
+            // the des-unique_ptr won't be used anywhere after this line.
+            handle_elaboration_errors();
+            return des->errors;
       }
 
       des->set_flags(flags);
 
       switch(min_typ_max_flag) {
-	case MIN:
-	    des->set_delay_sel(Design::MIN);
-	    break;
-	case TYP:
-	    des->set_delay_sel(Design::TYP);
-	    break;
-	case MAX:
-	    des->set_delay_sel(Design::MAX);
-	    break;
-	default:
-	    assert(0);
+      case MIN:
+            des->set_delay_sel(Design::MIN);
+            break;
+      case TYP:
+            des->set_delay_sel(Design::TYP);
+            break;
+      case MAX:
+            des->set_delay_sel(Design::MAX);
+            break;
+      default:
+            assert(0);
       }
 
-	/* Done with all the pform data. Delete the modules. */
+      /* Done with all the pform data. Delete the modules. */
       for (map<perm_string,Module*>::iterator idx = pform_modules.begin()
-		 ; idx != pform_modules.end() ; ++ idx ) {
+            ; idx != pform_modules.end() ; ++ idx ) {
 
-	    delete (*idx).second;
-	    (*idx).second = 0;
+            delete (*idx).second;
+            (*idx).second = 0;
       }
 
       if (verbose_flag) {
-	    if (times_flag) {
-		  times(cycles+2);
-		  cerr<<" ... done, "
-		      <<cycles_diff(cycles+2, cycles+1)<<" seconds."<<endl;
-	    }
-	    cout << "RUNNING FUNCTORS" << endl;
+            if (times_flag) {
+                  times(cycles+2);
+                  cerr<<" ... done, "
+                  <<cycles_diff(cycles+2, cycles+1)<<" seconds."<<endl;
+            }
+            cout << "RUNNING FUNCTORS" << endl;
       }
 
       while (!net_func_queue.empty()) {
-	    net_func func = net_func_queue.front();
-	    net_func_queue.pop();
-	    if (verbose_flag)
-		  cerr<<" -F "<<net_func_to_name(func)<< " ..." <<endl;
-	    func(des);
+            net_func func = net_func_queue.front();
+            net_func_queue.pop();
+            if (verbose_flag)
+            {
```

Comment:
This departs from the existing code style in two ways:
  - single statements aren't enclosed in braces
  - the opening brace is not put on a separate line

Reply:
Will fix the first point once I'm back from vacation in 2 weeks. Regarding your second point, there are [counterexamples](https://github.com/steveicarus/iverilog/blob/master/main.cc#L1291) in the existing code. I'd strongly prefer to use that style and be more explicit about what's inside and outside the if-statement. It helps avoid bugs imo. Since the style throughout the file is already inconsistent, we might as well make use of the advantages of being a little more explicit.
Are you sure you want me to change that?